src/components/RecipesList.js

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useEffect } from 'react';
+import RecipeService from '../services/RecipeService';
+import { Link } from 'react-router-dom';
+
+const RecipesList = () => {
+  const [recipes, setRecipes] = useState([]);
+  const [currentRecipe, setCurrentRecipe] = useState(null);
+  const [currentIndex, setCurrentIndex] = useState(-1);
+  const [searchName, setSearchName] = useState("");
+
+  useEffect(() => {
+    retrieveRecipes();
+  }, []);
+
+  const onChangeSearchName = event => {
+    const searchName = event.target.value;
+    setSearchName(searchName);
+  };
+
+  const retrieveRecipes = () => {
+    RecipeService.getAll()
+      .then(response => {
+        setRecipes(response.data);
+        console.log(response.data);
+      })
+      .catch(e => {
+        console.log(e);
+      });
+  };
+
+  const setActiveRecipe = (recipe, index) => {
+    setCurrentRecipe(recipe);
+    setCurrentIndex(index);
+  };
+
+  const findByName = () => {
+    RecipeService.findByName(searchName)
+      .then(response => {
+        setRecipes(response.data);
+        console.log(response.data);
+      })
+      .catch(e => {
+        console.log(e);
+      });
+  };
+
+  return (
+    <div className="list row">
+      <div className="col-md-8">
+        <div className="input-group mb-3">
+          <input
+            type="text"
+            className="form-control"
+            placeholder="Search by name"
+            value={searchName}
+            onChange={onChangeSearchName}
+            data-testid="recipes-search-bar"
+          />
+          <div className="input-group-append">
+            <button
+              className="btn btn-outline-secondary"
+              type="button"
+              onClick={findByName}
+              data-testid="recipes-search-btn"
+            >Search
+            </button>
+          </div>
+        </div>
+      </div>
+      <div className="col-md-6">
+        <h4 data-testid="recipes-header">Recipes List</h4>
+
+        <ul data-testid="recipes-listgroup" className="list-group">
+          {
+            recipes && recipes.map((recipe, index) => (
+              <li
+                data-testid={"recipe-" + recipe.id}
+                className={
+                  "list-group-item " + (index === currentIndex ? "active" : "")
+                }
+                onClick={() => setActiveRecipe(recipe, index)}
+                key={index}
```

Comment:
Let's always try to avoid using the index of an array as key as indicated on the [React docs](https://reactjs.org/docs/lists-and-keys.html#keys). In this case, it'd be better to use the id of the recipe.